posthog/hogql_queries/web_analytics/stats_table.py

```diff
@@ -438,6 +343,124 @@ def to_path_bounce_query(self) -> ast.SelectQuery:
         assert isinstance(query, ast.SelectQuery)
         return query
 
+    def _main_inner_query(self, breakdown, *, include_session_properties=False):
+        query = parse_select(
+            """
+SELECT
+    any(person_id) AS filtered_person_id,
+    count() AS filtered_pageview_count,
+    {breakdown_value} AS breakdown_value,
+    session.session_id AS session_id,
+    any(session.$is_bounce) AS is_bounce,
+    min(session.$start_timestamp) as start_timestamp
+FROM events
+WHERE and(timestamp >= {date_from}, timestamp < {date_to}, {event_where}, {all_properties}, {where_breakdown})
+GROUP BY session_id, breakdown_value
+""",
+            timings=self.timings,
+            placeholders={
+                "breakdown_value": breakdown,
+                "date_from": self._date_from_previous_period(),
+                "date_to": self._date_to(),
+                "event_where": self.event_type_expr,
+                "all_properties": self._all_properties(),
+                "where_breakdown": self.where_breakdown(),
+            },
+        )
+
+        assert isinstance(query, ast.SelectQuery)
+
+        if include_session_properties:
+            # Append session properties to the where clause if it exists
+            #
+            # We know `query.where` is an `ast.Call` because it's a `WHERE` clause
+            # and `ast.Call` is the only kind of expression that `query.where` can be
+            # but we need to convince mypy
+            if query.where is not None and isinstance(query.where, ast.Call):
+                query.where.args.append(self._session_properties())
+
+        if self.conversion_count_expr and self.conversion_person_id_expr:
+            query.select.append(ast.Alias(alias="conversion_count", expr=self.conversion_count_expr))
+            query.select.append(ast.Alias(alias="conversion_person_id", expr=self.conversion_person_id_expr))
+
+        return query
+
+    def _period_comparison_tuple(self, column, alias, function_name):
+        return ast.Alias(
+            alias=alias,
+            expr=ast.Tuple(
+                exprs=[
+                    self._current_period_aggregate(function_name, column),
+                    self._previous_period_aggregate(function_name, column),
+                ]
+            ),
+        )
+
+    def _current_period_aggregate(self, function_name, column_name):
+        return self.period_aggregate(function_name, column_name, self._date_from(), self._date_to())
+
+    def _previous_period_aggregate(self, function_name, column_name):
+        return self.period_aggregate(function_name, column_name, self._date_from_previous_period(), self._date_from())
+
+    # Reproduction from `web_analytics/web_overview.py`
```

Comment:
Could these functions move to the parent class to avoid duplication?

Reply:
I'll need to create a new parent class because `conversion_goal_expr` doesn't exist on the base query. I'll see how that goes, I agree it might be better than reproducing them here.

Deep down I'm ok with this because we'll eventually have a single query powering web analytics, and then there won't be any duplication anymore

Reply:
It was actually easy to extract this, done ✅